### PR TITLE
Add subcommands to parser

### PR DESCRIPTION
**Pull request by cogitantium**

### CODE REVIEW

**Comment by cogitantium** [comment]
I'm not sure if the return-type of functions `main` and `model_check` are correct. I believe some of the issues stems from having the main loop under the scope of the logger-handler.
Suggestions are welcome.

File: src/main.rs [hunk 1]
```diff
@@ -95,59 +125,83 @@ fn model_check(args: ArgMatches) -> Result<(), Box<dyn Error>> {
...
+                    .long("output")
+                    .env("OUTPUT")
+                    .help("The path to write output to"),
+            )
+    }
+
+    let mut parser = App::new(PKG_NAME)
+        .version(VERSION)
+        .author(AUTHORS);
+    parser = build_common_arguments(parser);
```
Comment (NicEastvillage) [inline src/main.rs:182]: This seems to be unused. Is that correct?

File: src/main.rs [hunk 2]
```diff
@@ -63,27 +63,57 @@ fn main() {
...
+    let game_structure = match args.value_of("model_type") {
+        Some("lcgs") => {
+            // TODO: implement lgcs parser
+            EagerGameStructure {
+                player_count: 0,
+                labeling: vec![],
+                transitions: vec![],
+                moves: vec![],
+            }
+        }
```
Comment (NicEastvillage) [inline src/main.rs:103]: You have already checked `model_type` in main. Why do it here again?

File: src/main.rs (hunk 1 above)
Comment (cogitantium) [reply src/main.rs:182]: Oops, old testing code.

File: src/main.rs (hunk 2 above)
Comment (cogitantium) [reply src/main.rs:103]: Because I was unsure where we'll make the lcgs parsing. Fixed in af880a9

File: src/main.rs (hunk 1 above)
Comment (cogitantium) [reply src/main.rs]: Removed in af880a9

**Review by NicEastvillage — APPROVED** [review]
The structure of main could be better to avoid `exit(1)`, but I will disregard it.

**Comment by cogitantium** [comment]
> The structure of main could be better to avoid `exit(1)`, but I will disregard it.

This was from a suggestion from @jenrik during #18, which I agree with.

**Comment by NicEastvillage** [comment]
We could also just return 🤷

**Review by MathiasMehl — APPROVED** [review]

**Comment by cogitantium** [comment]
I have removed the feature flags for graph-printing. I don't think it makes sense to have a parser with a graph subcommand and only maybe have the functionality enabled.

**Review by GeneralDenmark — APPROVED** [review]

**Pull request merged**